### PR TITLE
In simulator.py, I made a function that toggles a nonlocal boolean, and in the while running loop, I put an if boolean condition for the routes to be rendered.

### PR DESCRIPTION
**Pull request by horlikheng**

### CODE REVIEW

**Comment by CraftinPark** [comment]
Task linked: [CU-86b0rn6mq Toggle visibility of routes](https://app.clickup.com/t/86b0rn6mq)

**Review by changoug — DISMISSED** [review]
Good stuff!

File: src/simulator/simulator.py [hunk 1]
```diff
@@ -58,14 +63,14 @@ def restart_func() -> None:
...
-
         # fill the screen with a color to wipe away anything from last frame
         screen.fill("grey")
 
         render_buttons(screen, buttons)
 
         # optionally render nodes and edges. for now always on
-        render_world(screen, nodes, edges)
+        if route_visible:
+            render_world(screen, nodes, edges)
```
Comment (changoug) [inline src/simulator/simulator.py]: render_world may hold other components in the future. Consider passing a flag through render_world to toggle the routes specifically.

**Review by changoug — CHANGES_REQUESTED** [review]
Good work! render_world may contain other components in the future. Consider passing a flag into render_world instead.

**Review by changoug — APPROVED** [review]

**Pull request merged**